components/Dialogs/src/MDCDialogPresentationController.h

```diff
@@ -16,6 +16,18 @@
 
 #import "MaterialShadowElevations.h"
 
+@class MDCDialogPresentationController;
+
+/**
+ Similar to UIPopoverPresentationControllerDelegate, MDCDialogPresentationControllerDelegate lets
+ you customize the behavior of a popover-based presentation.
+ */
+@protocol MDCDialogPresentationControllerDelegate <NSObject>
+@optional
```

Comment:
Are you sure these should be optional?
If there is a single purpose to a protocol, then there is no reason to make its methods optional. Client apps that don't want to implement the methods can choose to simply not conform to the protocol.

Reply:
I think this is a matter of preference. I left it optional in case we add more methods later on and feel at that point that this one shouldn't be required. Do you want me to make it required @galiak11 ?

Reply:
+1 to making it required. It's my opinion that it's nicer to have many targeted delegates rather than a single big one with optional APIs.

Reply:
okie dokie, will change

Reply:
The `delegate` property pattern in Objective-C has a good deal of precedence for making its APIs optional. The reasoning being that there is typically a single `delegate` property, but likely many methods on that property that can optionally be implemented.

While I do agree that many small protocols with required methods are generally preferable (Swift encourages this, for example), the convention we're following here has a deep history in Objective-C of marking delegate APIs optional.

If we want to break from the `delegate` property historical precedence, then the `delegate` property name needs to be renamed to something that more specifically identifies its intended purpose. E.g. `didDismissDelegate`.

That being said, and given how much we rely on the `delegate` pattern in our code base already, I'm hesitant to encourage deviating from the established UIKit pattern and would like to encourage us to follow the UIKit conventions here of a `delegate` property with optional methods.

Reply:
Okie dokie, will make optional again